src/map/skill.c

```diff
@@ -8058,11 +8066,19 @@ static int skill_castend_nodamage_id(struct block_list *src, struct block_list *
 
 		case NPC_RUN:
 			{
-				const int mask[8][2] = {{0,-1},{1,-1},{1,0},{1,1},{0,1},{-1,1},{-1,0},{-1,-1}};
-				uint8 dir = (bl == src)?unit->getdir(src):map->calc_dir(src,bl->x,bl->y); //If cast on self, run forward, else run away.
+				enum unit_dir dir;
+				if (bl == src) //If cast on self, run forward, else run away.
+					dir = unit->getdir(src);
+				else
+					dir = map->calc_dir(src, bl->x, bl->y);
+				if (Assert_chk(dir >= UNIT_DIR_FIRST && dir < UNIT_DIR_MAX)) {
+					map->freeblock_unlock(); // unblock before assert-returning
+					return 0;
+				}
 				unit->stop_attack(src);
 				//Run skillv tiles overriding the can-move check.
-				if (unit->walktoxy(src, src->x + skill_lv * mask[dir][0], src->y + skill_lv * mask[dir][1], 2) && md)
+				if (unit->walk_toxy(src, (src->x + skill_lv * -dirx[dir]),
+				    (src->y + skill_lv * -diry[dir]), 2) == 0 && md != NULL)
```

Comment:
Redundant parenthesis.
```
if (unit->walk_toxy(src, src->x + skill_lv * -dirx[dir],
    src->y + skill_lv * -diry[dir], 2) == 0 && md != NULL)
```
Furthermore I would not break the line inside the function call here, since it drastically decreases the readability.
I'd prefer to only break `&& md != NULL` into a new line, even though the line length limit might gets broken then.

Reply:
Done.